### DISCUSSION TITLE
Allocate water resources in different spatiotemporal distributions

### DISCUSSION THREAD

**Discussion by luanshilaogao** [iiasa/CWatM#116] [Q&A] [answered]
Hello CWatM developers :
Thank you very much for taking the time to solve my problem amidst your busy schedule.
I recently wrote a water transfer code applicable to the CWatM model for optimizing water resource allocation. The code can adjust the variable 'unmetdemand' within a specified area given a total amount.
During this process, I attempted to allocate water resources in different spatiotemporal distributions, and in terms of time, I could allocate them on a monthly scale; But when encountering problems in space, how should I allocate water to a certain area? I noticed the 'usingAllocSegments' option, but in the model, it is only used in calculating' averageBaseflowInput '. The variable 'self. var. allocation_zone' also exists in the model, but it is only related to groundwater extraction. The function I hope to achieve is to provide an NC file containing IDs, each representing a different administrative region (configured according to this during water distribution), and the model uses this as a mask so that I can operate on this specific area.
Is it because I overlooked an option that could complete this function? If so, what should I do?

**Comment by luanshilaogao** [comment]
Hope for your reply！

**Comment by mikhailsmilovic** [comment, marked as answer]
Guten Tag luanshilaogao,
There are two main features for spatially allocating water availability: 
1. Using Input maps showing shared areas: Reservoir and lift command areas
2. Automatically creating shared zones: Zonal abstraction and Water body buffer

## Reservoir command areas (sharing reservoirs)
- Reservoir command areas are collections of cells that have access to the largest reservoir within the command area to satisfy demands.
- Command area maps identify cells with the same positive integer in the same command area. Cells outside a command area have a 0 value. 

In [OPTIONS]: 
**using_reservoir_command_areas** = True

In [WATERDEMAND}: 
**reservoir_command_areas** =  _path to raster/tif_

## Lift command areas (sharing channel storage)
- Lift command areas are cell collections that share channel storage to satisfy demands.  
- Command area maps identify cells with the same positive integer in the same command area. Cells outside a command area have a 0 value. 

In [OPTIONS]: 
**using_lift_areas** = True

In [WATERDEMAND}: 
**lift_areas** =  _path to raster/tif_

## Zonal abstraction (sharing channel storage and groundwater)
If in [OPTIONS]:
zonal_abstraction = True
limitAbstraction = True

Then, CWatM creates areas of shared channel storage and groundwater of cell radius **allocation_area**:
In [WATERDEMAND]:
**allocation_area** = _integer radius_

## Water body buffer (sharing lakes and reservoirs)
If in [OPTIONS]:
**using_reservoir_command_areas** = False (or **using_reservoir_command_areas** is not in the setting's file)

Then, CWatM creates buffers around lakes and reservoirs that can access the waterbody to satisfy water demands:
In [WATERDEMAND]:
**buffer_waterbodies** = _integer buffer_

**Reply by luanshilaogao** [reply]
I think so, Reservoir and lift command areas and Zonal abstraction respectively achieve uniform distribution of reservoirs, rivers, and groundwater, making the water supply process more realistic. A water source concentrated in several grid points can supply an entire area for use.
I hope to achieve a function similar to this, and for the water supply in this area, I hope it is a value I give. I have already written the program on how to allocate water, and the rest is to use the lift related code as a template to rebuild the regional water transfer module.
Thank you again for your enthusiastic help!

**Reply by mikhailsmilovic** [reply]
Sounds interesting! Looking forward to learning more about your water allocation program!
